sos/report/plugins/ceph_iscsi.py

```diff
@@ -0,0 +1,50 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, RedHatPlugin, UbuntuPlugin
+
+
+class CephISCSI(Plugin, RedHatPlugin, UbuntuPlugin):
+
+    short_desc = "CEPH iSCSI"
+    plugin_name = "ceph_iscsi"
+    profiles = ("storage", "virt")
+
+    packages = ("ceph-iscsi",)
+    services = ("rbd-target-api", "rbd-target-gw")
+
+    def setup(self):
+        self.add_copy_spec(
+            [
+                "/etc/tcmu/tcmu.conf",
+                "/var/log/ceph/ceph-client.iscsi.log",
```

Comment:
In updating some of the other ceph plugins, I've found that the `/var/log` locations have changed in newer versions (16+) of ceph. Most include the fsid, e.g. `/var/log/ceph/` to `/var/log/ceph/fsid/`.

Can we double check to make sure we're able to collect this log file from both newer and older versions?

Reply:
I dug into this but I'm not sure I'm following. Indeed, there is a particular path conversion with fsid in bind mount with `cephadm`, but from a process point of view the path is just `/var/log/ceph`, e.g. `/var/log/ceph/ceph-osd.0.log`. So not sure if that's the case for non-cephadm deployments.

```
docker inspect ceph-894b217c-8b2c-11ed-829c-930768227c30-iscsi-iscsi-cephadm-onylll | jq '.[].Mounts | .[] | select(.Destination=="/var/log")'
{
  "Type": "bind",
  "Source": "/var/log/ceph/894b217c-8b2c-11ed-829c-930768227c30",
  "Destination": "/var/log",
  "Mode": "z",
  "RW": true,
  "Propagation": "rprivate"
}
```

Reply:
If the collections needs to happen in a container, then you need to specify the container in the `add_copy_spec()`/`add_cmd_output()` calls with the `container` parameter (either the container name or the ID). 

Having the containers listed in the `containers` tuple just enables the plugin to run. As written, these collections would still happen directly on the host - so to copy them from the host you would need to use `/var/log/ceph/*/ceph-client.iscsi.log` or similar.